client/flags.go

```diff
@@ -4,11 +4,14 @@ import "github.com/spf13/cobra"
 
 // nolint
 const (
+	DefaultGasLimit = 200000
```

Comment:
Why do we need this at all? If the user doesn't want to simulate the transaction I think we should force them to manually specify gas.

Reply:
Makes sense, I will change it accordingly.
I was thinking to drop the `--gas-auto` flag in favor of having `--gas` value set to 0 by default, which would enable the auto gas feature.